SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -2087,145 +1546,777 @@ def gen_3d_survey_from_2d_lines(
-    z = np.hstack(z)
-    z = z + mesh.x0[0]
-    if xlim == None:
-        rho_min = rho[~np.isnan(rho)].min() * 0.5
-        rho_max = rho[~np.isnan(rho)].max() * 2
-    else:
-        rho_min, rho_max = xlim
-
-    if xscale == "linear" and rho.min() == 0.0:
-        if xlim == None:
-            rho_min = -rho[~np.isnan(rho)].max() * 0.5
-            rho_max = rho[~np.isnan(rho)].max() * 2
-
-    if ax == None:
-        plt.xscale(xscale)
-        plt.xlim(rho_min, rho_max)
-        plt.ylim(z.min(), z.max())
-        plt.xlabel("Resistivity ($\Omega$m)", fontsize=14)
-        plt.ylabel("Depth (m)", fontsize=14)
-        plt.ylabel("Depth (m)", fontsize=14)
-        if showlayers:
-            for locz in z_grid:
-                plt.plot(
-                    np.linspace(rho_min, rho_max, 100),
-                    np.ones(100) * locz,
-                    "b--",
-                    lw=0.5,
-                )
-        return plt.plot(resistivity, z, "k-", **kwargs)
 
-    else:
-        ax.set_xscale(xscale)
-        ax.set_xlim(rho_min, rho_max)
-        ax.set_ylim(z.min(), z.max())
-        ax.set_xlabel("Resistivity ($\Omega$m)", fontsize=14)
-        ax.set_ylabel("Depth (m)", fontsize=14)
-        if showlayers:
-            for locz in z_grid:
-                ax.plot(
-                    np.linspace(rho_min, rho_max, 100),
-                    np.ones(100) * locz,
-                    "b--",
-                    lw=0.5,
-                )
-        return ax.plot(resistivity, z, "k-", **kwargs)
 
 
 ############
 # Deprecated
 ############
 
-
-def plot_pseudoSection(
+def plot_pseudosection(
     data,
     ax=None,
     survey_type="dipole-dipole",
-    data_type="appConductivity",
-    space_type="half-space",
+    data_type="apparent conductivity",
+    space_type="half space",
+    plot_type="pcolor",
     clim=None,
     scale="linear",
     sameratio=True,
-    pcolorOpts={},
-    data_location=False,
+    pcolor_opts={},
+    contour_opts={},
+    cbar_opts={},
+    data_locations=False,
     dobs=None,
     dim=2,
+    y_values="n-spacing",
+    **kwargs,
 ):
 
     warnings.warn(
-        "The plot_pseudoSection method has been deprecated. Please use "
-        "plot_pseudosection instead. This will be removed in version"
+        "The plot_pseudosection method has been deprecated and may not work properly!!! "
+        "Please use plot_2d_pseudosection instead. This will be removed in version"
         " 0.15.0 of SimPEG",
         DeprecationWarning,
     )
 
-    return plot_pseudosection(
-        data=data,
-        ax=ax,
-        survey_type=survey_type,
-        data_type=data_type,
-        space_type=space_type,
-        clim=clim,
-        scale=scale,
-        sameratio=sameratio,
-        pcolor_opts=pcolorOpts,
-        data_locations=data_location,
-        dobs=dobs,
-        dim=dim,
+    return plot_2d_pseudosection(data.survey, data.dobs, 'scatter', ax=ax, scale=scale)
```

Comment:
there's a lot of options that are not passed on to the `plot_pseudosection` function (i.e. pcolor_opts)